crates/dojo/bindgen/src/plugins/unrealengine/mod.rs

```diff
@@ -0,0 +1,1376 @@
+            CArrayStruct* models = &entities->data[i].models;
+            this->ParseModelsAndSend(models);
+        }}
+        FDojoModule::CArrayFree(entities->data, entities->data_len);
+    }});
+    }}
+
+void AGeneratedHelpers::SubscribeOnDojoModelUpdate()
+{{
+    UE_LOG(LogTemp, Log, TEXT(\"Subscribing to entity update.\"));
+    if (subscribed) {{
+        UE_LOG(LogTemp, Log, TEXT(\"Warning: cancelled, already subscribed.\"));
+        return;
+    }}
+    if (toriiClient == nullptr) {{
+        UE_LOG(LogTemp, Log, TEXT(\"Error: Torii Client is not initialized.\"));
+        return;
+    }}
+    subscribed = true;
+    struct ResultSubscription res =
+        FDojoModule::OnEntityUpdate(toriiClient, \"{{}}\", nullptr, CallbackProxy);
+    subscription = res.ok;
+}}
+
+void AGeneratedHelpers::CallbackProxy(struct FieldElement key, struct CArrayStruct models)
+{{
+    if (!Instance) return;
+    Instance->ParseModelsAndSend(&models);
+}}
+
+template<typename T>
+static void ConvertTyToUnrealEngineType(const Member* member, const char* expectedName, const \
+             char* expectedType, T& output);
+
+{class_types_converter}
+
+{function_parse_models}
+{calls_functions_bodies}
+",
+            contract_addresses = contract_addresses,
+            policies = policies,
+            class_types_converter = class_types_converter,
+            function_parse_models = function_parse_models,
+            calls_functions_bodies = calls_functions_bodies
+        );
+
+        out
+    }
+}
+
+fn check_token_in_recursively(token: &Token, type_name: &str) -> bool {
+    match token {
+        Token::Composite(composite) => {
+            if composite.type_name() == type_name {
+                return true;
+            }
+            composite.inners.iter().any(|inner| check_token_in_recursively(&inner.token, type_name))
+        }
+        Token::Array(array) => check_token_in_recursively(&array.inner, type_name),
+        Token::Tuple(tuple) => {
+            tuple.inners.iter().any(|inner| check_token_in_recursively(inner, type_name))
+        }
+        _ => token.type_name() == type_name,
+    }
+}
+
+#[async_trait]
+impl BuiltinPlugin for UnrealEnginePlugin {
+    async fn generate_code(&self, data: &DojoData) -> BindgenResult<HashMap<PathBuf, Vec<u8>>> {
+        let mut out: HashMap<PathBuf, Vec<u8>> = HashMap::new();
+
+        let mut models = data.models.iter().collect::<Vec<_>>();
+        // Sort models based on their tag to ensure deterministic output.
+        models.sort_by(|(_, a), (_, b)| a.tag.cmp(&b.tag));
+
+        let mut contracts = data.contracts.iter().collect::<Vec<_>>();
+        // Sort contracts based on their tag to ensure deterministic output.
+        contracts.sort_by(|(_, a), (_, b)| a.tag.cmp(&b.tag));
+
+        println!("Generating header: GeneratedHelpers.h");
+        let code = self.handle_header(&models, &contracts);
+        out.insert("GeneratedHelpers.h".into(), code.as_bytes().to_vec());
```

Comment:
Shouldn't be easier to find with `DojoHelpers`? Or it's a convention in unreal engine?